ecosystem/gstreamer_plugin/gstmtltxsink.c

```diff
@@ -150,6 +158,63 @@ static gboolean gst_mtltxsink_parse_input_fmt(GstVideoInfo* info,
   return TRUE;
 }
 
+static gboolean gst_mtltxsink_parse_input_fmt_code(const char* format, enum st_frame_fmt* fmt) {
+  if (!fmt || !format) {
+    GST_ERROR("%s, invalid input\n", __func__);
+    return FALSE;
+  }
+
+  if (strcmp(format, "YUV422PLANAR10LE") == 0) {
```

Comment:
Is it not worthy to check first the length of the "format" variable (string) and then compare with appropriate strings according to this length only? It would be more time optimal ate least.